src/Engine/Component/GeometryComponent.cpp

```diff
@@ -217,59 +223,70 @@ void TriangleMeshComponent::setupIO( const std::string& id ) {
     ComponentMessenger::getInstance()->registerOutput<Ra::Core::Utils::Index>(
         getEntity(), this, id, roOut );
 
-    ComponentMessenger::CallbackTypes<Ra::Core::Vector3Array>::ReadWrite vRW =
-        std::bind( &TriangleMeshComponent::getVerticesRw, this );
-    ComponentMessenger::getInstance()->registerReadWrite<Ra::Core::Vector3Array>(
-        getEntity(), this, id + "v", vRW );
-
-    ComponentMessenger::CallbackTypes<Ra::Core::Vector3Array>::ReadWrite nRW =
-        std::bind( &TriangleMeshComponent::getNormalsRw, this );
-    ComponentMessenger::getInstance()->registerReadWrite<Ra::Core::Vector3Array>(
-        getEntity(), this, id + "n", nRW );
-
-    ComponentMessenger::CallbackTypes<TriangleArray>::ReadWrite tRW =
-        std::bind( &TriangleMeshComponent::getTrianglesRw, this );
-    ComponentMessenger::getInstance()->registerReadWrite<TriangleArray>(
-        getEntity(), this, id + "t", tRW );
+    // ComponentMessenger::CallbackTypes<
+    //     Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>::ReadWrite vRW =
+    //     std::bind( &TriangleMeshComponent::getVerticesRw, this );
+    // ComponentMessenger::getInstance()
+    //     ->registerReadWrite<Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>(
+    //         getEntity(), this, id + "v", vRW );
+
+    // ComponentMessenger::CallbackTypes<
+    //     Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>::ReadWrite nRW =
+    //     std::bind( &TriangleMeshComponent::getNormalsRw, this );
+    // ComponentMessenger::getInstance()
+    //     ->registerReadWrite<Ra::Core::Geometry::TriangleMesh::PointAttribHandle::AttribType>(
+    //         getEntity(), this, id + "n", nRW );
+
+    //    ComponentMessenger::CallbackTypes<TriangleArray>::ReadWrite tRW =
+    //        std::bind( &TriangleMeshComponent::getTrianglesRw, this );
+    //    ComponentMessenger::getInstance()->registerReadWrite<TriangleArray>(
+    //        getEntity(), this, id + "t", tRW );
```

Comment:
The Plugin from https://gitlab.com/Storm-IRIT/radium-official-plugins/skeletonbasedanimation uses the `ReadWrite` callbacks to modify the skinned mesh. Can you propose a PR on it about how to do so without these callbacks?

Reply:
Sure, I just have to find how to fork and PR on gitlab, but I already have adapted the plugin code to work with the new behavior (i.e. get the mesh, and then access with lock to the data, not the data directly)

Reply:
You did so on the example plugins (github), but not on the official ones (gitlab)